src/Microsoft.Net.Http.Headers/SetCookieHeaderValue.cs

```diff
@@ -342,15 +341,15 @@ private static int GetSetCookieLength(string input, int startIndex, out SetCooki
                     }
                     var dateString = ReadToSemicolonOrEnd(input, ref offset);
                     DateTimeOffset expirationDate;
-                    if (!HttpRuleParser.TryStringToDate(dateString, out expirationDate))
+                    if (!HttpRuleParser.TryStringToDate(dateString.ToString(), out expirationDate))
```

Comment:
I think we should just change it to be `StringSegment` and just call ToString() in the appropriate places.